### PR TITLE
Optimize TMB calculation

### PR DESCRIPTION
**Pull request by hassanfa**
### This PR:

Updates:

- Improves TMB calculation for panelseq data

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: BALSAMIC/snakemake_rules/annotation/vep.rule [hunk 1]
```diff
@@ -90,11 +90,14 @@ source activate {params.conda};
 
 if [ \"{params.bed}\" == \"\" ]; then region_size=3101.78817; else region_size=$(awk '{{s+=$3-$2}}END{{print s/1e6}}' {params.bed}); fi;
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/annotation/vep.rule:91]:
```suggestion
if [ \"{params.bed}\" == \"\" ]; then region_size=3101.78817; 
```
For panels we always have the bed file right ? Then what is the purpose of pre-assigning this value `3101.78817` when bed file is none ?

**Review by ashwini06 — COMMENTED** [review]
👍

**Review by khurrammaqbool — APPROVED** [review]

**Comment by hassanfa** [comment]
@khurrammaqbool @ashwini06 

TMB calculation is following the method described in the linked issue: https://github.com/Clinical-Genomics/BALSAMIC/issues/51

It is optimized according to this publication as well: https://pubmed.ncbi.nlm.nih.gov/30238975/

For that weird number, have a look at for total genome length and why it is a float number: https://www.ncbi.nlm.nih.gov/grc/human/data?asm=GRCh37.p13

Also note that TMB is not validated yet, but this is an effort to optimize it further until we do some sort of validation.

File: BALSAMIC/snakemake_rules/annotation/vep.rule [hunk 2]
```diff
@@ -75,7 +75,7 @@ rule vep_stat:
     Path(singularity_image, config["bioinfo_tools"].get("ensembl-vep") + ".sif").as_posix() 
   params:
     conda = config["bioinfo_tools"].get("ensembl-vep"),
-    bcftools_filter_string = "INFO/DP>=50 && FORMAT/AD>=5 && FORMAT/AF>=0.1",
```
[the remark this answers is not in the document]
Comment (hassanfa) [reply BALSAMIC/snakemake_rules/annotation/vep.rule:78]: See the comment. There are two publications on how to calculate tmb. TMB is still away from being complete, as AF cutoff should be applied anyway. See the publications and issue in my latest comment.

File: BALSAMIC/snakemake_rules/annotation/vep.rule (hunk 1 above)
Comment (khurrammaqbool) [reply BALSAMIC/snakemake_rules/annotation/vep.rule:91]: Correction:  Genome size/100000

Comment (hassanfa) [reply BALSAMIC/snakemake_rules/annotation/vep.rule:91]: Divided by 1e6. Total length is 3 101 788 170, thus 3101.78817. Again this is a very crude estimation. The actual size should be calculated from the reference we are using. At least until we validate TMB.

**Pull request merged**